driver-core/src/test/functional/com/mongodb/internal/connection/AsyncSocketChannelStreamSpecification.groovy

```diff
@@ -70,4 +73,47 @@ class AsyncSocketChannelStreamSpecification extends Specification {
         then:
         thrown(MongoSocketOpenException)
     }
+
+    @IgnoreIf({ getSslSettings().isEnabled() })
+    def 'should fail AsyncCompletionHandler if name resolution fails'() {
+        given:
+        def socketSettings = SocketSettings.builder().connectTimeout(100, MILLISECONDS).build()
+        def channelGroup = AsynchronousChannelGroup.withThreadPool(Executors.newFixedThreadPool(5))
```

Comment:
This threadpool isn't cleaned up and probably isn't needed as the test could just create the AsynchronousSocketChannelStream directly.